src/ralph/utils.py

```diff
@@ -37,23 +55,60 @@ def import_string(dotted_path):
         ) from err
 
 
-def get_backend_type(backends: BaseModel, backend_name: str):
+def get_backend_type(backend_types: List[BaseModel], backend_name: str):
     """Return the backend type from a backend name."""
     backend_name = backend_name.upper()
-    for _, backend_type in backends:
+    for backend_type in backend_types:
         if hasattr(backend_type, backend_name):
             return backend_type
     return None
 
 
-def get_backend_instance(backend_type: BaseModel, backend_name: str, options: dict):
-    """Return the instantiated backend instance given backend-name-prefixed options."""
+def get_backend_class(backend_type: BaseModel, backend_name: str):
+    """Return the backend class given the backend type and backend name."""
+    # Get type name from backend_type class name
+    backend_type_name = backend_type.__class__.__name__[
+        : -len("BackendSettings")
+    ].lower()
+    backend_name = backend_name.lower()
```

Comment:
I believe this function and/or doc should be clarified in the interest of future developments. I find it pretty hard to follow.

In particular, it is confusing to have: `backend_type`, `backend_name`, and `backend_type_name` in addition to `backend_class`.

Reply:
Opened an issue https://github.com/openfun/ralph/issues/457 so we can improve clarity in the future